Source/WebCore/platform/graphics/displaylists/DisplayListResourceHeap.h

```diff
@@ -92,15 +92,14 @@ class ResourceHeap {
         add<FontCustomPlatformData>(renderingResourceIdentifier, WTFMove(customPlatformData), m_customPlatformDataCount);
     }
 
-    ImageBuffer* getImageBuffer(RenderingResourceIdentifier renderingResourceIdentifier) const
+    RefPtr<ImageBuffer> getImageBuffer(RenderingResourceIdentifier renderingResourceIdentifier) const
```

Comment:
I do not think you need to change the return type form a raw pointer to a RefPtr. The caller side will create a RefPtr implicitly if the type of the received variable is a RefPtr.

Reply:
The main issue is that some of these do a `dynamicDowncast`, which clang annoyingly doesn't seem to understand should have the same lifetime as its source (as long as it's not modified), so I stuffed the downcast pointer into a `RefPtr` to explicitly add a reference after the function ends.
I may have done this one and other non-downcasting ones for consistency, I could revert these. Or do you object to all `RefPtr` returns? Other suggestions? (And I'll try to think of alternatives.)